### ISSUE TITLE
Add AddServerComponents() to Blazor Web App project template

### ISSUE THREAD

**Issue by TonyHenrique** [dotnet/aspnetcore#48952]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the solution you'd like

I have been using blazor for some years, but now I tried the new Blazor Web App project template and noticed that a very important call is missing:
`services.AddRazorComponents().AddServerComponents()`

I think that this should be included, or at least commented, in the project template:

```
services.AddRazorComponents()
//.AddServerComponents() // Enables interactivity
;
```

to make easier for new users to quickly make things work easily as it was on the previous template.

**Comment by TonyHenrique** [comment]
Hi, @javiercn ! I checked again now and used both `dotnet new` command  and Microsoft Visual Studio Community 2022 (64-bit) - Preview (Version 17.7.0 Preview 2.0)
and got this same template that does not even mention the `.AddServerComponents()`
```
dotnet --list-sdks
6.0.100-preview.7.21379.14 [C:\Program Files\dotnet\sdk]
6.0.410 [C:\Program Files\dotnet\sdk]
7.0.400-preview.23274.1 [C:\Program Files\dotnet\sdk]
8.0.100-preview.5.23303.2 [C:\Program Files\dotnet\sdk]

dotnet new blazor
O modelo "Blazor Web App" foi criado com êxito.
Este modelo contém tecnologias de outras pessoas além da Microsoft, consulte https://aka.ms/aspnetcore/8.0-third-party-notices para obter detalhes.

Processando ações pós-criação...
Restaurando C:\temp\blazor_net8\blazor_net8.csproj:
  Determinando os projetos a serem restaurados...
  C:\temp\blazor_net8\blazor_net8.csproj restaurado (em 154 ms).
A restauração foi bem-sucedida.

```

![image](https://github.com/dotnet/aspnetcore/assets/6307623/04d4c345-6036-4790-a284-f597c33bc38e)
![image](https://github.com/dotnet/aspnetcore/assets/6307623/b7ead6e7-2c15-4520-b873-7647efb4f8f6)
![image](https://github.com/dotnet/aspnetcore/assets/6307623/10001308-17a9-45e8-bcc4-f0ecdb009731)

Program.cs

```
using BlazorApp3;

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.
builder.Services.AddRazorComponents();

var app = builder.Build();

// Configure the HTTP request pipeline.
if (!app.Environment.IsDevelopment())
{
    app.UseExceptionHandler("/Error");
    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
    app.UseHsts();
}

app.UseHttpsRedirection();

app.UseStaticFiles();

app.MapRazorComponents<App>();

app.Run();
```

**Comment by javiercn** [comment]
@TonyHenrique we have yet to surface the server option on the template.

It's there, just hidden. It'll be visible in preview6.

**Comment by TonyHenrique** [comment]
Got it.

Then, please consider this to be included, or at least commented, in the project template:

```
services.AddRazorComponents()
//.AddServerComponents() // Enables blazor server interactivity
;
```

**Comment by javiercn** [comment]
@TonyHenrique why is it not enough with the option in the template?

**Comment by TonyHenrique** [comment]
I think that the whole idea is converging to have a **single Blazor project template** that works seamlessy on server, webassembly and now also has the kind of static behaviour like a classic ASP.NET of PHP page, wich is very cool.

So to me it does not make sense to not include/hide the option that makes Blazor interactivity work out of the box and require me to create another project with some option to have this included.

For example, I am used to create a new blazor project, hit the Visual Studio play button and the interactivity works. Now when trying the new template, I got surprised that I ran the new project and the interactivity did not work, so I had to manually search for and add that call `AddServerComponents()`. 

That is why I suggested that at least it came commented on the template code.
